PSICompile/ILCodeGen.cs

```diff
@@ -66,9 +85,34 @@ public class ILCodeGen : Visitor {
       }
       if (w.NewLine) Out ("    call void [System.Console]System.Console::WriteLine ()");
    }
-   
-   public override void Visit (NIfStmt f) => throw new NotImplementedException ();
-   public override void Visit (NForStmt f) => throw new NotImplementedException ();
+
+   public override void Visit (NIfStmt f) {
+      string lab1 = NextLabel (), lab2 = NextLabel (), lab3 = NextLabel ();
+      Out ($"    br {lab2}");
+      Out ($"  {lab1}:"); f.IfPart.Accept (this); Out ($"    br {lab3}");
+      Out ($"  {lab2}:"); f.Condition.Accept (this); 
+      Out ("    ldc.i4.1\n    ceq");
+      Out ($"    brtrue {lab1}");
+      if (f.ElsePart != null) f.ElsePart.Accept (this);
+      Out ($"  {lab3}:"); Out ("  nop");
+   }
+   public override void Visit (NForStmt f) {
+      string lab1 = NextLabel (), lab2 = NextLabel ();
+      f.Start.Accept (this); StoreVar (f.Var);
+      Out ($"    br {lab2}");
+      Out ($"  {lab1}:");
+      f.Body.Accept (this);
+      Visit (new NIdentifier (f.Var));
+      Out ($"    ldc.i4 1");
+      Out (f.Ascending ? "    add" : "    sub");
+      StoreVar (f.Var);
+      Out ($"  {lab2}:");
+      Visit (new NIdentifier (f.Var));
+
+      f.End.Accept (this);

```

Comment:
If you swap these two statements thus:
```
f.End.Accept (this);
Visit (new NIdentifier (f.Var));
```
your relational check on the next line will be simpler. Also, see if you can avoid creating an NIdentifier object here.

Reply:
Implemented LoadVar method

Reply:
Simplied the relational check